pkg/controller/azvolume.go

```diff
@@ -286,31 +285,29 @@ func (r *ReconcileAzVolume) triggerDelete(ctx context.Context, azVolume *azdiskv
 
 				// wait async and report error to go channel
 				wg.Add(1)
-				go func(ctx context.Context, waiter *watcher.ConditionWaiter, i int) {
+				go func(ctx context.Context, waiter *watcher.ConditionWaiter, attachment azdiskv1beta2.AzVolumeAttachment) {
 					defer waiter.Close()
 					defer wg.Done()
 					_, err := waiter.Wait(ctx)
 					if err != nil {
-						errors[i] = err
-						atomic.AddUint32(&numErrors, 1)
+						errorMessageCh <- fmt.Sprintf("%s: %v", attachment.Name, err)
 					}
-				}(deleteCtx, waiter, i)
+				}(deleteCtx, waiter, attachment)
 			}
 
 			// start waiting for the attachment clean up to complete (outside of the (if err == nil) to avoid child workflow finishing after parent workflow finishing)
 			wg.Wait()
+			close(errorMessageCh)
 
 			if err == nil {
-				// if errors have been found with the wait calls, format the error msg and report via CRI
-				if numErrors > 0 {
+				go func(errorMessageCh chan string) {
```

Comment:
This shouldn't be done in a goroutine as below would have dependency on this operation.

https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/0c86bb7e168a42435da2211cef0535f2d48f76ea/pkg/controller/azvolume.go#L315

Reply:
Done.